eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -311,42 +312,53 @@ private[journal] object ReplicatedCassandra {
                               .as(journalHead.some)
                           } { journalHead =>
                             if (offset > journalHead.partitionOffset.offset) {
-                              val deleteTo0 = journalHead.deleteTo
-                              val seqNr     = journalHead.seqNr
-                              val seqNr1    = deleteTo0.fold { seqNr } { _.value.max(seqNr) }
-                              val deleteTo1 = deleteTo
+                              val headDeleteTo = journalHead.deleteTo
+                              val headSeqNr    = journalHead.seqNr
+
+                              // keep existing `seqNr` or update to higher one
+                              // usually because `delete batch` removed previous `append batch`
+                              val validatedSeqNr: SeqNr = setSeqNr.fold { headSeqNr } { _.max(headSeqNr) }
+
+                              // clamp down to requested or max valid seqNr
+                              // in default protocol we have `Journal#delete(DeleteTo.max)`
+                              // `DeleteTo.max` can be used to discard all traces, but keep `seqNr`
+                              val validatedDeleteTo = deleteTo
                                 .value
-                                .min(seqNr1)
+                                .min(validatedSeqNr)
                                 .toDeleteTo
+
+                              val updateSeqNr = setSeqNr.isDefined
```

Comment:
And one more remark, more important than one above:

_Before the change_, code in `def delete` was also responsible for side-function - fix meta where
```
journalHead.deleteTo > journalHead.seqNr
```

Lets follow this cherry-picked lines _before the change_:
```
                              val deleteTo0 = journalHead.deleteTo
                              val seqNr       = journalHead.seqNr
                              val seqNr1     = deleteTo0.fold { seqNr } { _.value.max(seqNr) }
                              
                              if (seqNr1 == seqNr) { ... } else {
                                   update(seqNr1, ...)
                              }
```

As far as I see, _after the change_ `metajournal.seq_nr` will be updated only if 
```
setSeqNr.isDefined
```

Reply:
I am not sure that it is correct to "overload" the `Delete` action with extra functionality like:
> fix the case where somehow `delete_to > seq_nr` in `metajournal` table

It would mean that we do not trust our code to do the update correctly - I checked that in both client and replicator parts we clamp down `DeleteTo` to head's `seqNr`:
* in client: https://github.com/evolution-gaming/kafka-journal/blob/ba2031dd853b8f6d08b2db6d2e5e5d920cf9cfd3/journal/src/main/scala/com/evolutiongaming/kafka/journal/Journals.scala#L308
* in replicator: https://github.com/evolution-gaming/kafka-journal/blob/ba2031dd853b8f6d08b2db6d2e5e5d920cf9cfd3/eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala#L309